File: crates/hyper_render/src/scene.rs
//
// Copyright (c) 2024, SkillerRaptor
//
// SPDX-License-Identifier: MIT
//

use hyper_ecs::Registry;

pub struct Scene {
    _registry: Registry,
}

impl Scene {
    pub fn new() -> Self {
        Self {
            _registry: Registry::new(),
        }
    }
}

File: hyper_rendering/src/hyper_rendering/scene.cpp
/*
 * Copyright (c) 2025-present, SkillerRaptor
 *
 * SPDX-License-Identifier: MIT
 */

#include "hyper_rendering/scene.hpp"

namespace hyper_engine
{
    entt::registry &Scene::registry()
    {
        return m_registry;
    }

    const entt::registry &Scene::registry() const
    {
        return m_registry;
    }
} // namespace hyper_engine